src/user/user.service.ts

```diff
@@ -391,4 +393,43 @@ export class UserService {
             throw new BadRequestException("Bad confirmation token");
         }
     }
+
+    public async resendEmailVerification(
+        hashedEmail: string
+    ): Promise<boolean> {
+        try {
+            const user = await this.userRepository.findOneBy({
+                email: hashedEmail
+            });
+            if (!user || !user.publicKey)
+                throw new Error("Email or PGP key not found");
+
+            const publicKey = await openpgp.readKey({
+                armoredKey: user.publicKey
+            });
+
+            const emailFromPublicKey = publicKey.users[0].userID.email;
+
+            if (hashMessage(emailFromPublicKey) != hashedEmail)
+                throw new Error("Email does not match pgp key");
+
+            const token =
+                this._generateEmailVerificationToken(emailFromPublicKey);
+            await this.emailService.sendEmailVerification(
+                emailFromPublicKey,
+                token
+            );
+            return true;
+        } catch (error) {
+            this.logger.error(error);
+            throw new Error(error);
+        }
+    }
+
+    private _generateEmailVerificationToken(
+        emailFromPublicKey: string
+    ): string {
+        const payload = { email: emailFromPublicKey };
+        return this.jwtService.sign(payload);
```

Comment:
why not just
`return this.jwtService.sign({ email: emailFromPublicKey})`
or even better, rename the arg to `email` and do `return this.jwtService.sign({ email})`

Reply:
Because the email pass in is hashedEmail